### PR TITLE
Fix deprecated use of ReflectionParameter::getClass for php 8

### PR DESCRIPTION
**Pull request by cacahouwete**

### CODE REVIEW

**Comment by cacahouwete** [comment]
@pamil is it possible to review this PR ?

**Comment by stronk7** [comment]
FYI, we have applied the patch here to our (hopefully temporal) fork and it's working ok, so far. No more PHP Deprecated warnings.

Only point that called my attention are the changes back-slashing global scope functions (`\class_exists()...) that doesn't seem to be the rule followed by other parts of MinkExtension... just saying.

File: src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ public function configure(ArrayNodeDefinition $builder)
      */
     public function buildDriver(array $config)
     {
-        if (!class_exists('Behat\Mink\Driver\GoutteDriver')) {
+        if (!\class_exists('Behat\Mink\Driver\GoutteDriver')) {
```
Comment (andrewnicols) [inline src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php:65]: No need to prefix these with \ They are already globally scoped.

File: src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php [hunk 2]
```diff
@@ -125,7 +126,7 @@ private function isGoutte1()
 
     private function isGuzzle6()
     {
-        return interface_exists('GuzzleHttp\ClientInterface') &&
-            version_compare(\GuzzleHttp\ClientInterface::VERSION, '6.0.0', '>=');
+        return \interface_exists('GuzzleHttp\ClientInterface') &&
```
Comment (andrewnicols) [inline src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php:129]: No need to prefix these with \ They are already globally scoped.

**Review by andrewnicols — CHANGES_REQUESTED** [review]
These changes look good to me except for the use of \ to globally scope functions which are in the global scope.

File: src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php (hunk 2 above)
Comment (cacahouwete) [reply src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php]: Done

File: src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php (hunk 1 above)
Comment (cacahouwete) [reply src/Behat/MinkExtension/ServiceContainer/Driver/GoutteFactory.php]: Done

**Comment by cacahouwete** [comment]
@andrewnicols is it possible to merge it ?

**Comment by andrewnicols** [comment]
@cacahouwete , nothing to do with me. This repository is the responsibility of FriendsOfBehat I just gave a review as we have included this pull request in our packages.

Pinging @lchrusciel , @pamil , and @Zales0123 - this issues blocks support for PHP 8.0 and is therefore critical for most.

**Comment by acsauk** [comment]
Hey @pamil - is there a likelihood of this being merged soon? It's causing some failures in our behat suites following a PHP 8 upgrade. If it's not on the cards are you able to suggest any alternative approaches to the package? Thanks.

**Review by Yozhef — APPROVED** [review]
thanks for the help we will be glad to see you again

**Pull request merged**